Objects/listobject.c

```diff
@@ -253,6 +253,21 @@ PyList_GetItem(PyObject *op, Py_ssize_t i)
     return ((PyListObject *)op) -> ob_item[i];
 }
 
+PyObject *
+PyList_GetItemRef(PyObject *op, Py_ssize_t i)
+{
+    if (!PyList_Check(op)) {
+        PyErr_SetString(PyExc_TypeError, "expected a list");
+        return NULL;
+    }
+    if (!valid_index(i, Py_SIZE(op))) {
+        _Py_DECLARE_STR(list_err, "list index out of range");
```

Comment:
Does it work to call _Py_DECLARE_STR() twice? What if the message changes later in a future commit? I suggest to only call it once.

Maybe _Py_DECLARE_STR() should be declared at the "module level" (outside the function) to show that it's "shared". The implementation is trivial:

```
#define _Py_DECLARE_STR(name, str)
```

Reply:
The documentation for `_Py_DECLARE_STR` says:

```
_Py_DECLARE_STR() should precede all uses of _Py_STR() in a function.

   This is true even if the same string has already been declared
   elsewhere, even in the same file.  Mismatched duplicates are detected
   by Tools/scripts/generate-global-objects.py.
```

Reply:
Ah thank you, I didn't read the doc, it's my fault.